include/mapbox/variant_io.hpp

```diff
@@ -20,7 +21,14 @@ class printer
 
     // visitor
     template <typename T>
-    void operator()(T const& operand) const
+    typename std::enable_if<std::is_empty<T>::value, void>::type operator()(T const&) const
+    {
+        out_ << "[]";
```

Comment:
Is this driven by json spec ^ ? The syntax implies an empty array, maybe this OK but worth checking

Reply:
On second look, this is definitely not OK to force '[]' for all "empty" types.

Reply:
Fair enough - I think I can solve the issue I have differently - I'm going to abandon this PR for now.